x-pack/plugins/security_solution/public/common/components/toolbar/bulk_actions/alert_bulk_tags.tsx

```diff
@@ -68,35 +88,37 @@ const BulkAlertTagsPanelComponent: React.FC<BulkAlertTagsPanelComponentProps> =
       if (refresh) refresh();
       if (clearSelection) clearSelection();
     };
-    if (setAlertTags != null) {
-      setAlertTags(tags, ids, onSuccess, setIsLoading);
+    if (onSubmit != null) {
+      closePopoverMenu();
+      await onSubmit(tags, ids, onSuccess, setIsLoading);
     }
   }, [
     closePopoverMenu,
     tagsToAdd,
     tagsToRemove,
     alertItems,
-    setAlertTags,
     refetchQuery,
     refresh,
     clearSelection,
     setIsLoading,
+    onSubmit,
   ]);
 
-  const handleTagsOnChange = (
-    newOptions: EuiSelectableOption[],
-    event: EuiSelectableOnChangeEvent,
-    changedOption: EuiSelectableOption
-  ) => {
-    if (changedOption.checked === 'on') {
-      tagsToAdd.add(changedOption.label);
-      tagsToRemove.delete(changedOption.label);
-    } else if (!changedOption.checked) {
-      tagsToRemove.add(changedOption.label);
-      tagsToAdd.delete(changedOption.label);
-    }
-    setSelectableAlertTags(newOptions);
-  };
+  const handleTagsOnChange = useCallback(
+    (
+      newOptions: EuiSelectableOption[],
+      event: EuiSelectableOnChangeEvent,
+      changedOption: EuiSelectableOption
+    ) => {
+      if (changedOption.checked === 'on') {
+        addAlertTag(changedOption.label);
+      } else if (!changedOption.checked) {
```

Comment:
Would we ever have the third in between state? The one where we show the `-` in the UI? I noticed clicking through the tags when doing bulk tags that I can't get it back to the state of `-`, it just goes between checked and unchecked for all, no unset.

Reply:
Yeah, that's built in as part of the EUI component. The thought would be there's no use case to go from none -> some or all -> some. As an intermediate state within the selectable dropdown, it only exists as an initial setting.